tests/rptest/tests/log_segment_ms_test.py

```diff
@@ -0,0 +1,196 @@
+SERVER_SEGMENT_MS = 60000
+SERVER_HOUSEKEEPING_LOOP = 10
+
+
+def messages_to_generate_segments(num_segments, period):
+    return int(period / 1000 * (num_segments) + SERVER_HOUSEKEEPING_LOOP + 1)
+
+
+TEST_NUM_SEGMENTS = 3
+ERROR_MARGIN = 1
+TEST_LOG_SEGMENT_MIN = 60000
+TEST_LOG_SEGMENT_MAX = 150000
+
+
+class SegmentMsTest(RedpandaTest):
+    def __init__(self, test_context):
+        super().__init__(test_context=test_context, num_brokers=1)
+
+    def _total_segments_count(self, topic_spec: TopicSpec, partition=0):
+        return next(segments_count(self.redpanda, topic_spec.name, partition))
+
+    def generate_workload(self,
+                          server_cfg: typing.Optional[int],
+                          topic_cfg: typing.Optional[int],
+                          use_alter_cfg: bool,
+                          num_messages: int,
+                          extra_cluster_cfg={}) -> tuple[int, int]:
+        """
+        Create some workload to trigger time-based segment rolling. setup cluster, create a topic with config
+        (either at creation time or with alter_config),
+        then starts a producer at 1hz of small #num_messages.
+        returns the number of segment before and after the workload
+        """
+        self.redpanda.set_cluster_config({'log_segment_ms': server_cfg}
+                                         | extra_cluster_cfg,
+                                         expect_restart=False)
+
+        topic = TopicSpec()
+        rpk = RpkTool(self.redpanda)
+        rpk.create_topic(topic.name,
+                         partitions=1,
+                         config={TopicSpec.PROPERTY_SEGMENT_MS: topic_cfg} if
+                         topic_cfg is not None and not use_alter_cfg else None)
+
+        if use_alter_cfg:
+            if topic_cfg:
+                rpk.alter_topic_config(topic.name,
+                                       TopicSpec.PROPERTY_SEGMENT_MS,
+                                       topic_cfg)
+            else:
+                rpk.delete_topic_config(topic.name,
+                                        TopicSpec.PROPERTY_SEGMENT_MS)
+
+        producer = VerifiableProducer(context=self.test_context,
+                                      num_nodes=1,
+                                      redpanda=self.redpanda,
+                                      topic=topic.name,
+                                      max_messages=num_messages,
+                                      throughput=1)
+        consumer = VerifiableConsumer(context=self.test_context,
+                                      num_nodes=1,
+                                      redpanda=self.redpanda,
+                                      topic=topic.name,
+                                      group_id=0)
+        start_count = self._total_segments_count(topic)
+        consumer.start()
+        producer.start()
+        wait_until(lambda: producer.num_acked >= num_messages,
```

Comment:
Why is it necessary to use a wait_until on num_acked?  If the intent is to produce a set number of messages, then you could set `max_messages` on the producer and then just `wait()` on it.